test/Sentinel1MetadataTest.jl

```diff
@@ -0,0 +1,180 @@
+    else
+        return isXML
+    end
+end
+
+
+
+
+
+#############################################
+### test structures and constructurs #######
+### excluding the MetaDataSentinel1 struct #######
+#############################################
+
+
+
+function HeaderTest()
+    metaDict = Sentinel1.getDictofXml(xmlFile)
+    header = Sentinel1.Header(metaDict)
+    #testing if data exists in header
+    checkTimes = header.startTime != nothing
+    checkTypes = typeof(header.startTime) == DateTime && typeof(header.stopTime) == DateTime
+    check = checkTimes && checkTypes
+    if !check
+        println("Error in Header")
+        println("Start time ", header.startTime)
+        println("Stop time: ", header.stopTime)
+    end
+    return check
+end
+
+
+function productInformationTest()
+
+    metaDict = Sentinel1.getDictofXml(xmlFile)
+    product = Sentinel1.ProductInformation(metaDict)
+    ## Assert
+    checkTypes = typeof(product.rangeSamplingRate) == Float64
+    checkrangeSamplingRate = product.rangeSamplingRate > 0
+    checkTypesProduct2 = product.radarFrequency > 5 #frq should be 5.4 Ghz ish
+
+
+    check = checkrangeSamplingRate && checkTypes && checkTypesProduct2
+    if !check 
+        println("Error in Product data")
+        println("Samplig rate", product.rangeSamplingRate, "of type ", typeof(product.rangeSamplingRate))
+    end
+    return check
+end
+
+
+function ImageInformationTest()
+
+    metaDict = Sentinel1.getDictofXml(xmlFile)
+    imageinfo = Sentinel1.ImageInformation(metaDict)
+    ## Assert
+    checkTypesImage2 = imageinfo.azimuthFrequency > 50 #frequency shoul be is 485 Hz ish
+    if !checkTypesImage2
+        println("Error in Image data")
+    end
+    return checkTypesImage2
+end
+
+
+
+function GeolocationGridTest()
+
+    metaDict = Sentinel1.getDictofXml(xmlFile)
+    geolocation = Sentinel1.GeolocationGrid(metaDict)
+    ## Assert
+
+
+    checkGeolocation1 = minimum(geolocation.lines) > 0
+    checkGeolocation2 = minimum(geolocation.longitude) > -181
+    checkGeolocation3 = minimum(geolocation.latitude) > -181
+    checkGeolocation4 = minimum(geolocation.height) > 0
+    checkGeolocation5 = minimum(geolocation.elevationAngle) > 0
+
+
+    check = checkGeolocation1 && checkGeolocation2 && checkGeolocation3 && checkGeolocation4 && checkGeolocation5
+    if !check
+        println("Error in GeolocationGridTest")
+        println(checkGeolocation1)
+        println(checkGeolocation2)
+        println(checkGeolocation3)
+        println(checkGeolocation4)
+        println(checkGeolocation5)
+    end
+    return check
+end
+
+
+function BurstTest()
+    #Action
+    metaDict = Sentinel1.getDictofXml(xmlFile)
+    burstinfo = Sentinel1.BurstsInfo(metaDict)
+
+    azimuthTime = [burst.azimuthTime for burst in burstinfo.bursts]
+    lastValidSample = [burst.lastValidSample for burst in burstinfo.bursts]
+
+    ## Assert
+    # checkTypes
+    checkLength = length(azimuthTime) == length(lastValidSample) 
```

Comment:
Maybe just hardcode the number of burst in the test file and check that the length fits with that.
I can see that they are equal length but are they the right length?

Reply:
Actually do some checks of the values. The ´length(azimuthTime) == length(lastValidSample)´  does not really check anything in the code other than it runs.

Reply:
Added a few checks for this specific testfile (SLC).